base/acme/src/main/java/org/dogtagpki/acme/database/LDAPDatabase.java

```diff
@@ -356,17 +359,35 @@ public void importSchema(LDAPConnection connection) throws Exception {
         }
     }
 
-    public void createIndexes(LDAPConnection connection) throws Exception {
+    public void createIndexes(LDAPConnection connection, Map<String, String> params) throws Exception {
 
         logger.info("Creating ACME indexes");
 
+        String backend = params.getOrDefault("backend", "userroot");
+
         String filename = "/usr/share/pki/acme/database/ds/index.ldif";
         LDIF ldif = new LDIF(filename);
 
         while (true) {
             LDIFRecord record = ldif.nextRecord();
             if (record == null) break;
 
+            // parse cn=<name>,cn=index,cn=<backend>,cn=ldbm database,cn=plugins,cn=config
```

Comment:
The `index.ldif` file has no explicit `cn=<backend>` in the DNs. Should the file be modified to use that value instead of `userroot` to make evident that it will be replaced with the real value?